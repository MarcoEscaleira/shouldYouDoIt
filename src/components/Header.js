import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header className="header">
    <nav className="header__nav">
      <NavLink to="/" exact className="nav__title">
        Should You <span className="nav__title__span">Do It</span>
      </NavLink>
      <ul className="nav__ul">
        <li className="nav__ul__li">
          <NavLink to="/" exact className="navLink" activeClassName="isActive">
            Home
          </NavLink>
        </li>
        <li className="nav__ul__li">
          <NavLink to="/gallery" className="navLink" activeClassName="isActive">
            Gallery
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
