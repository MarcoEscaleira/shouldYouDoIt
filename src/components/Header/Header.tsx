import React from "react";
import {
  Container,
  Nav,
  List,
  Item,
  Link,
  TitleSpan
} from "./styled-components";

const Header: React.FC = () => (
  <Container>
    <Nav>
      <Link to="/" exact isTitle>
        Should You <TitleSpan>Do It</TitleSpan>
      </Link>
      <List>
        <Item>
          <Link to="/" exact isNavLink activeClassName="activeHeaderLink">
            Home
          </Link>
        </Item>
        <Item>
          <Link to="/gallery" isNavLink activeClassName="activeHeaderLink">
            Gallery
          </Link>
        </Item>
      </List>
    </Nav>
  </Container>
);

export default Header;
