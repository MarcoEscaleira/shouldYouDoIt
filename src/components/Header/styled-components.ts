// @ts-nocheck
import styled, { css } from "styled-components";
import { NavLink } from "react-router-dom";
import {
  pallete,
  sizes,
  normalFont,
  headerHeight,
  headerDesktopHeight
} from "~/styles/_settings";

export const Container = styled.header`
  width: 100%;
  font-family: ${normalFont};
  background-color: rgba(29, 154, 108, 0.3);
  padding: 0 1.5rem;
  height: ${headerHeight};

  @media (min-width: ${sizes.desktop_breakpoint}) {
    height: ${headerDesktopHeight};
  }
`;

export const Nav = styled.nav`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-transform: uppercase;
  font-size: ${sizes.m};
`;

export const List = styled.ul`
  display: flex;
`;

export const Item = styled.li`
  list-style: none;
`;

type HeaderLink = {
  isTitle?: boolean;
  isNavLink?: boolean;
};

const linkBorder = `${pallete.fifth_light} 0.1rem solid`;

export const Link = styled(NavLink)<HeaderLink>`
  ${({ isTitle }) =>
    isTitle &&
    css`
      font-size: 1.6rem;
      font-weight: 300;
      color: ${pallete.fifth_light};

      @media (min-width: ${sizes.desktop_breakpoint}) {
        font-size: 2.2rem;
      }
    `}
  ${({ isNavLink }) =>
    isNavLink &&
    css`
    font-size: ${sizes.s}
    margin: 0 1rem;
    padding: 0.6rem 0;
    letter-spacing: 0.2rem;
    color: ${pallete.lightest};

    &:hover {
      border-top: ${linkBorder};
      border-bottom: ${linkBorder};
    }

    @media (min-width: ${sizes.desktop_breakpoint}) {
      font-size: 1.3rem;
    }
  `}

  &:last-of-type {
    margin-right: 0;
  }
`;

export const TitleSpan = styled.span`
  color: ${pallete.second_light};
  font-weight: 600;
`;
