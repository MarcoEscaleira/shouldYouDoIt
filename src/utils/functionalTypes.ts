export type NullaryFn<R> = () => R;
export type UnaryFn<A, R> = (a: A) => R;
export type BinaryFn<A, B, R> = (a: A, b: B) => R;
export type TernaryFn<A, B, C, R> = (a: A, b: B, c: C) => R;
export type QuadFn<A, B, C, D, R> = (a: A, b: B, c: C, d: D) => R;
export type QuinFn<A, B, C, D, F, R> = (a: A, b: B, c: C, d: D, f: F) => R;
export type SixFn<A, B, C, D, F, H, R> = (
  a: A,
  b: B,
  c: C,
  d: D,
  f: F,
  h: H
) => R;
