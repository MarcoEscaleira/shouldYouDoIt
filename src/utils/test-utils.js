import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import store from '../store';

export const renderWithRedux = (ui, initialState) => {
  return {
    ...render(
      <Provider store={store(initialState)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = ui => {
  return {
    ...render(<Router>{ui}</Router>)
  };
};

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
