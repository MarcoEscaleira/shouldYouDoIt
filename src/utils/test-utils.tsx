import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import store, { initialStoreState } from "../store";

export const renderWithRedux = (ui: JSX.Element, initialState?: {}) => {
  const state = {
    ...initialStoreState,
    ...initialState
  };
  return {
    ...render(
      <Provider store={store(state)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = (ui: JSX.Element) => ({
  ...render(<Router>{ui}</Router>)
});

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
