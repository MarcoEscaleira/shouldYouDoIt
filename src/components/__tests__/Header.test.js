import React from 'react';
import { renderWithRouter } from '../../utils/test-utils';
import Header from '../Header';

describe('Header test suite', () => {
  it('Should render Header correctly', () => {
    const { container } = renderWithRouter(<Header />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
