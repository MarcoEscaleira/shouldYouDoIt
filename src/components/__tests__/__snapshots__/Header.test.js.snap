// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test suite Should render Header correctly 1`] = `
<header
  class="header"
>
  <nav
    class="header__nav"
  >
    <a
      aria-current="page"
      class="nav__title active"
      href="/"
    >
      Should You 
      <span
        class="nav__title__span"
      >
        Do It
      </span>
    </a>
    <ul
      class="nav__ul"
    >
      <li
        class="nav__ul__li"
      >
        <a
          aria-current="page"
          class="navLink isActive"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="nav__ul__li"
      >
        <a
          class="navLink"
          href="/gallery"
        >
          Gallery
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
